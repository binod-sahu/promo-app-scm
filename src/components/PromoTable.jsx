import React from "react";
import PromoItems from './PromoItems'
import config from '../appConfig'

const PromoTable = () => {
    return (
      <div className="container">
        <table className="table caption-top">
            <caption className="text-center"><h1>Promo Table</h1></caption>
            <thead className="table-dark">
                <tr>
                <th scope="col">No.</th>
                <th scope="col">SKU ID</th>
                <th scope="col">Quantity</th>
                <th scope="col">Base Price</th>
                <th scope="col">Total Price</th>
                <th scope="col">Discount</th>
                <th scope="col">Payable Amount</th>
                </tr>
            </thead>
            <tbody>
              { config.map((item, index) =>
              <PromoItems 
                item= {item}
                index={index}
                key={index}
              />)
              }  
              
            </tbody>
            </table>  
      </div>
    );
  }
  
  export default PromoTable;