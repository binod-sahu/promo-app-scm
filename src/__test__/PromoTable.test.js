import React from "react";
import { render, fireEvent, act } from '@testing-library/react'
import PromoTable from "../components/PromoTable";
import renderer from 'react-test-renderer';

describe("snapshot", () => {
  const props = {
    itemName: "A",
    price: 50,
    count: 0,
    discount: 0,
    total: 0,
    promoEngine: jest.fn()
  };
  test("should render snapshot", () => {
    const renderedModule = renderer.create(<PromoTable />).toJSON();;
    expect(renderedModule).toMatchSnapshot();
  });

  test("should render snapshot1 with props", () => {
    const renderedModule = renderer.create(<PromoTable {...props} />).toJSON();;
    expect(renderedModule).toMatchSnapshot();
  });
});
