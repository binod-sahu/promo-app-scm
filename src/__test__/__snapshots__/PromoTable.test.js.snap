// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot should render snapshot 1`] = `
<div
  className="container"
>
  <table
    className="table caption-top"
  >
    <caption
      className="text-center"
    >
      <h1>
        Promo Table
      </h1>
    </caption>
    <thead
      className="table-dark"
    >
      <tr>
        <th
          scope="col"
        >
          No.
        </th>
        <th
          scope="col"
        >
          SKU ID
        </th>
        <th
          scope="col"
        >
          Quantity
        </th>
        <th
          scope="col"
        >
          Base Price
        </th>
        <th
          scope="col"
        >
          Total Price
        </th>
        <th
          scope="col"
        >
          Discount
        </th>
        <th
          scope="col"
        >
          Payable Amount
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <th
          scope="row"
        >
          1
        </th>
        <td>
          A
        </td>
        <td
          className="d-flex"
        >
          <div
            className="col-auto my-1 me-2"
          >
            <button
              className="btn btn-primary"
              id="decrement"
              onClick={[Function]}
            >
              -
            </button>
          </div>
          <input
            className="form-control select-width me-2"
            onChange={[Function]}
            type="text"
            value={0}
          />
          <div
            className="my-1"
          >
            <button
              className="btn btn-primary"
              id="increment"
              onClick={[Function]}
            >
              +
            </button>
          </div>
        </td>
        <td>
          £
          50
        </td>
        <td>
          £
          0
        </td>
        <td>
          £
          0
        </td>
        <td>
          £
          0
        </td>
      </tr>
      <tr>
        <th
          scope="row"
        >
          2
        </th>
        <td>
          B
        </td>
        <td
          className="d-flex"
        >
          <div
            className="col-auto my-1 me-2"
          >
            <button
              className="btn btn-primary"
              id="decrement"
              onClick={[Function]}
            >
              -
            </button>
          </div>
          <input
            className="form-control select-width me-2"
            onChange={[Function]}
            type="text"
            value={0}
          />
          <div
            className="my-1"
          >
            <button
              className="btn btn-primary"
              id="increment"
              onClick={[Function]}
            >
              +
            </button>
          </div>
        </td>
        <td>
          £
          30
        </td>
        <td>
          £
          0
        </td>
        <td>
          £
          0
        </td>
        <td>
          £
          0
        </td>
      </tr>
      <tr>
        <th
          scope="row"
        >
          3
        </th>
        <td>
          C
        </td>
        <td
          className="d-flex"
        >
          <div
            className="col-auto my-1 me-2"
          >
            <button
              className="btn btn-primary"
              id="decrement"
              onClick={[Function]}
            >
              -
            </button>
          </div>
          <input
            className="form-control select-width me-2"
            onChange={[Function]}
            type="text"
            value={0}
          />
          <div
            className="my-1"
          >
            <button
              className="btn btn-primary"
              id="increment"
              onClick={[Function]}
            >
              +
            </button>
          </div>
        </td>
        <td>
          £
          20
        </td>
        <td>
          £
          0
        </td>
        <td>
          £
          0
        </td>
        <td>
          £
          0
        </td>
      </tr>
      <tr>
        <th
          scope="row"
        >
          4
        </th>
        <td>
          D
        </td>
        <td
          className="d-flex"
        >
          <div
            className="col-auto my-1 me-2"
          >
            <button
              className="btn btn-primary"
              id="decrement"
              onClick={[Function]}
            >
              -
            </button>
          </div>
          <input
            className="form-control select-width me-2"
            onChange={[Function]}
            type="text"
            value={0}
          />
          <div
            className="my-1"
          >
            <button
              className="btn btn-primary"
              id="increment"
              onClick={[Function]}
            >
              +
            </button>
          </div>
        </td>
        <td>
          £
          15
        </td>
        <td>
          £
          0
        </td>
        <td>
          £
          0
        </td>
        <td>
          £
          0
        </td>
      </tr>
    </tbody>
    <tfoot>
      <tr>
        <td
          className="fw-bolder"
          colSpan="4"
        >
          Total
        </td>
        <td
          className="fw-bold"
        >
          £
          0
        </td>
        <td
          className="fw-bold"
        >
          £
          0
        </td>
        <td
          className="fw-bold"
        >
          £
          0
        </td>
      </tr>
    </tfoot>
  </table>
  <div
    className="row text-end"
  >
    <div
      className="fs-4 fw-bold"
    >
      Total Price :
      <span
        className="fw-light fs-5 ms-2"
      >
        £
        0
      </span>
    </div>
    <div
      className="fs-4 fw-bold"
    >
      Total Discount :
      <span
        className="fw-light fs-5 ms-2"
      >
        £
        0
      </span>
    </div>
    <div
      className="fs-4 fw-bold"
    >
      Total Amount to pay :
      <span
        className="fw-light fs-5 ms-2"
      >
        £
        0
      </span>
    </div>
  </div>
</div>
`;

exports[`snapshot should render snapshot1 with props 1`] = `
<div
  className="container"
>
  <table
    className="table caption-top"
  >
    <caption
      className="text-center"
    >
      <h1>
        Promo Table
      </h1>
    </caption>
    <thead
      className="table-dark"
    >
      <tr>
        <th
          scope="col"
        >
          No.
        </th>
        <th
          scope="col"
        >
          SKU ID
        </th>
        <th
          scope="col"
        >
          Quantity
        </th>
        <th
          scope="col"
        >
          Base Price
        </th>
        <th
          scope="col"
        >
          Total Price
        </th>
        <th
          scope="col"
        >
          Discount
        </th>
        <th
          scope="col"
        >
          Payable Amount
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <th
          scope="row"
        >
          1
        </th>
        <td>
          A
        </td>
        <td
          className="d-flex"
        >
          <div
            className="col-auto my-1 me-2"
          >
            <button
              className="btn btn-primary"
              id="decrement"
              onClick={[Function]}
            >
              -
            </button>
          </div>
          <input
            className="form-control select-width me-2"
            onChange={[Function]}
            type="text"
            value={0}
          />
          <div
            className="my-1"
          >
            <button
              className="btn btn-primary"
              id="increment"
              onClick={[Function]}
            >
              +
            </button>
          </div>
        </td>
        <td>
          £
          50
        </td>
        <td>
          £
          0
        </td>
        <td>
          £
          0
        </td>
        <td>
          £
          0
        </td>
      </tr>
      <tr>
        <th
          scope="row"
        >
          2
        </th>
        <td>
          B
        </td>
        <td
          className="d-flex"
        >
          <div
            className="col-auto my-1 me-2"
          >
            <button
              className="btn btn-primary"
              id="decrement"
              onClick={[Function]}
            >
              -
            </button>
          </div>
          <input
            className="form-control select-width me-2"
            onChange={[Function]}
            type="text"
            value={0}
          />
          <div
            className="my-1"
          >
            <button
              className="btn btn-primary"
              id="increment"
              onClick={[Function]}
            >
              +
            </button>
          </div>
        </td>
        <td>
          £
          30
        </td>
        <td>
          £
          0
        </td>
        <td>
          £
          0
        </td>
        <td>
          £
          0
        </td>
      </tr>
      <tr>
        <th
          scope="row"
        >
          3
        </th>
        <td>
          C
        </td>
        <td
          className="d-flex"
        >
          <div
            className="col-auto my-1 me-2"
          >
            <button
              className="btn btn-primary"
              id="decrement"
              onClick={[Function]}
            >
              -
            </button>
          </div>
          <input
            className="form-control select-width me-2"
            onChange={[Function]}
            type="text"
            value={0}
          />
          <div
            className="my-1"
          >
            <button
              className="btn btn-primary"
              id="increment"
              onClick={[Function]}
            >
              +
            </button>
          </div>
        </td>
        <td>
          £
          20
        </td>
        <td>
          £
          0
        </td>
        <td>
          £
          0
        </td>
        <td>
          £
          0
        </td>
      </tr>
      <tr>
        <th
          scope="row"
        >
          4
        </th>
        <td>
          D
        </td>
        <td
          className="d-flex"
        >
          <div
            className="col-auto my-1 me-2"
          >
            <button
              className="btn btn-primary"
              id="decrement"
              onClick={[Function]}
            >
              -
            </button>
          </div>
          <input
            className="form-control select-width me-2"
            onChange={[Function]}
            type="text"
            value={0}
          />
          <div
            className="my-1"
          >
            <button
              className="btn btn-primary"
              id="increment"
              onClick={[Function]}
            >
              +
            </button>
          </div>
        </td>
        <td>
          £
          15
        </td>
        <td>
          £
          0
        </td>
        <td>
          £
          0
        </td>
        <td>
          £
          0
        </td>
      </tr>
    </tbody>
    <tfoot>
      <tr>
        <td
          className="fw-bolder"
          colSpan="4"
        >
          Total
        </td>
        <td
          className="fw-bold"
        >
          £
          0
        </td>
        <td
          className="fw-bold"
        >
          £
          0
        </td>
        <td
          className="fw-bold"
        >
          £
          0
        </td>
      </tr>
    </tfoot>
  </table>
  <div
    className="row text-end"
  >
    <div
      className="fs-4 fw-bold"
    >
      Total Price :
      <span
        className="fw-light fs-5 ms-2"
      >
        £
        0
      </span>
    </div>
    <div
      className="fs-4 fw-bold"
    >
      Total Discount :
      <span
        className="fw-light fs-5 ms-2"
      >
        £
        0
      </span>
    </div>
    <div
      className="fs-4 fw-bold"
    >
      Total Amount to pay :
      <span
        className="fw-light fs-5 ms-2"
      >
        £
        0
      </span>
    </div>
  </div>
</div>
`;
